### PR TITLE
Update package libgpg error

### PR DESCRIPTION
**Pull request by JSquar**
- awk is a build depdency for this package

There is an [issue](https://dev.gnupg.org/rE7865041c77f4f7005282f10f9b6666b19072fbdf) with libgpg-error and gawk >= 5.0 which will be fixed in any release version of libgpg-error (it's the third commit on their master branch after the release tag 1.36). Therefore there should be a requirement in the sense of 

    depends_on('awk@:4, when='@:1.36', type='build')
    depends_on('awk@4:, when='@1.37:', type='build')

But since awk is a virtual dependency this would not work I assume? Shall I leave it as it is or specify this version constraint for each awk provider?

### CODE REVIEW

**Comment by adamjstewart** [comment]
You could add a patch for that particular commit so the latest version works with any version of awk.

**Comment by adamjstewart** [comment]
Ping @JSquar

**Comment by JSquar** [comment]
Oh...er...thanks for the reminder :) I'll push the patch on the weekend.

File: var/spack/repos/builtin/packages/libgpg-error/package.py [hunk 1]
```diff
@@ -19,3 +19,6 @@ class LibgpgError(AutotoolsPackage):
     version('1.27', sha256='4f93aac6fecb7da2b92871bb9ee33032be6a87b174f54abf8ddf0911a22d29d2')
     version('1.21', sha256='b7dbdb3cad63a740e9f0c632a1da32d4afdb694ec86c8625c98ea0691713b84d')
     version('1.18', sha256='9ff1d6e61d4cef7c1d0607ceef6d40dc33f3da7a3094170c3718c00153d80810')
+
+    depends_on('awk', type=('build'))
+    patch('awk-5.patch', when='@1.36^gawk@5:')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/libgpg-error/package.py]: If we always use this patch, do we still need `depends_on('awk')`?

Comment (JSquar) [reply var/spack/repos/builtin/packages/libgpg-error/package.py]: This patch does not eliminate the need for awk, it just considers changes to the gawk interface. Without the dependency spack would use the globally installed awk implementation

**Pull request merged**